internal/test_cases/history_persistence_test_case.go

```diff
@@ -0,0 +1,114 @@
+package test_cases
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/codecrafters-io/shell-tester/internal/assertions"
+	"github.com/codecrafters-io/shell-tester/internal/logged_shell_asserter"
+	"github.com/codecrafters-io/shell-tester/internal/shell_executable"
+	"github.com/codecrafters-io/tester-utils/logger"
+)
+
+// HistoryPersistenceTestCase checks that after loading history from a file, the shell's history output matches the file's contents.
+type HistoryPersistenceTestCase struct {
+	// ExpectHistoryRCommand controls whether to expect a history -r command in the history
+	ExpectHistoryRCommand bool
```

Comment:
We can ship with this but let's pair and refactor this - the interface & implementation for this test case feels overly complex